File: 674.longest-continuous-increasing-subsequence.java
/*
 * @lc app=leetcode id=674 lang=java
 *
 * [674] Longest Continuous Increasing Subsequence
 */

// @lc code=start
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        int maxSequenceLength = 1;
        int currSequenceLength = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                currSequenceLength++;
            } else {
                maxSequenceLength = Math.max(maxSequenceLength, currSequenceLength);
                currSequenceLength = 1;
            }
        }

        return Math.max(maxSequenceLength, currSequenceLength);
    }
}
// @lc code=end

File: 674.longest-continuous-increasing-subsequence.py
#
# @lc app=leetcode id=674 lang=python3
#
# [674] Longest Continuous Increasing Subsequence
#

# @lc code=start
class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        curr_sequence_length = 1
        max_sequence_length = 1

        for i in range(1, len(nums)):
            
            if nums[i] > nums[i - 1]:
                curr_sequence_length += 1
            else:
                max_sequence_length = max(max_sequence_length, curr_sequence_length)
                curr_sequence_length = 1
        
        return max(max_sequence_length, curr_sequence_length)
                
# @lc code=end

